ccpp/driver/GFS_diagnostics.F90

```diff
@@ -40,17 +40,58 @@ module GFS_diagnostics
     real(kind=kind_phys) :: cnvfac
     type(data_subtype), dimension(:), allocatable :: data
    end type GFS_externaldiag_type
-
+   
   !--- public data type ---
   public  GFS_externaldiag_type
 
   !--- public interfaces ---
   public  GFS_externaldiag_populate
- 
+
   CONTAINS
 
 !%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-
+    
```

Comment:
Great to have this in its own subroutine. If we decide to put all the `GFS_typedefs` logic into its own module, could we move that subroutine, too? Unless it depends on imports in `GFS_diagnostics.F90`.

Reply:
As far as I know, all of the dtend & dtidx code in GFS_typedefs.F90 and GFS_diagnostics.F90 could be moved to a separate module, other than the type definitions. However, I'd rather not add new files until the upstream repository maintainers say they want more files.